### DISCUSSION TITLE
How to change gltf/PbrMetallicRoughness default Blend Mode to transparent ?

### DISCUSSION THREAD

**Discussion by ThanosRestas** [atteneder/glTFast#392] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/5999342/166449693-a2d887db-1daa-4424-8116-a56e431b434e.png)

Any input on how to do this will be deeply appreciated

**Comment by atteneder** [comment]
Can you elaborate in what kind of workflow you'd want transparent to be default?

I'm asking because:

- For the classical glTF import workflow (design- and run-time) the glTF material dictates the blend mode anyways
- For manually creating materials with that shader Opaque makes much more sense as default. Also, why would you want to manually use these Shaders?

Thanks for your input

**Reply by ThanosRestas** [reply]
Thanks for the reply ! So:

> Can you elaborate in what kind of workflow you'd want transparent to be default?

I want the transparency of the transparent textures to be visible on the glTF models I load. 

> For the classical glTF import workflow (design- and run-time) the glTF material dictates the blend mode anyways

What I'm basically asking is how to edit this glTF material so it comes as transparent by default. 

> For manually creating materials with that shader Opaque makes much more sense as default. Also, why would you want to manually use these Shaders?

I don't want to edit them manually(via the editor or whatever). I would like to find how you are generating this material so I can change its  blend mode to transparent via code at time of the model's instantiation. Like how would the Blend Mode parameter _just_ be set to transparent ?

Obviously I'm just starting to learn about how materials/shaders are created and applied so probably I'm confusing you a bit. Might be something I'm missing in understanding your code too.

**Reply by atteneder** [reply]
Alright, got you. I'll create a new answer.

**Comment by atteneder** [comment, marked as answer]
glTF comes with out-of-the box support for:

- Opaque materials
- Alpha Masking (equivalent to Blend Mode `Cutout`)
- Alpha Blending (equivalent to Blend Mode `Fade`)

That does not allow you to re-create materials like transparent glass, where light transmits, but there's still a specular reflection.

To achieve this, you have to make use of the [KHR_materials_transmission extension](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_materials_transmission/README.md). It allows you to create glass-like materials (even tinted or frosted ones). The shader's blend mode will implicitly be set to `transparent` if you use this extension. Make sure your glTF generator has proper support for it.

Note: Depending on your render pipeline (settings), glTFast only offers [partial support for transmission](https://github.com/atteneder/glTFast/blob/main/Documentation%7E/features.md). See #111 as well.
